janitor/functions.py

```diff
@@ -3869,3 +3813,230 @@ def sort_naturally(
     """
     new_order = index_natsorted(df[column_name], **natsorted_kwargs)
     return df.iloc[new_order, :]
+
+
+@pf.register_dataframe_method
+def expand_grid(df: pd.DataFrame = None,
+                df_key: str = None,
+                others: Dict = None) -> pd.DataFrame:
+    """ Creates a dataframe from all combinations of all inputs.
+        This method mutates the original dataframe.
+
+    Functional usage syntax:
+
+    .. code-block:: python
+
+        import pandas as pd
+        import janitor as jn
+
+        df = pd.DataFrame(...)
+        df = jn.expand_grid(df=df, df_key="...", others={...})
+
+    Method-chaining usage syntax:
+
+        import pandas as pd
+        import janitor as jn
+
+    .. code-block:: python
+
+        df = pd.DataFrame(...).expand_grid(df_key="bla",others={...})
+
```

Comment:
Because the function can be used independent of a dataframe, can you add in one syntax example?

```suggestion

    Usage independent of a DataFrame:

        .. code-block:: python
        import pandas as pd
        from janitor import expand_grid

        df = expand_grid(...please-fill-in-here...)
```